src/app-layer-parser.c

```diff
@@ -1011,6 +1014,20 @@ int AppLayerParserParse(ThreadVars *tv, AppLayerParserThreadCtx *alp_tctx, Flow
         }
     }
 
+    if (AppLayerParserProtocolIsTxAware(f->proto, alproto)) {
+        if (likely(tv)) {
+            /* At each pass in the parser we add to the tv counter
```

Comment:
@inliniac on the tx_cnt being part of the Flow structure, we have tried to explain it in this comment.